test/compilable/test12558.d

```diff
@@ -0,0 +1,38 @@
+// REQUIRED_ARGS:
+/*
+TEST_OUTPUT:
+---
+compilable/test12558.d(16): Deprecation: catch statement without an exception specification is deprecated; use catch(Throwable) for old behavior
+compilable/test12558.d(21): Deprecation: catch statement without an exception specification is deprecated; use catch(Throwable) for old behavior
+---
+*/
+
+void main()
+{
+    auto handler = () { };
+
+    try {
+        assert(0);
+    } catch
+        handler();
+
+    try {
+        assert(0);
+    } catch {
+        handler();
+    }
+
+    version (none)
+    {
+        try {
+            assert(0);
+        } catch  // should not emit diagnostics
```

Comment:
Original question: https://github.com/dlang/dmd/pull/3482#issuecomment-64800112
Answer: https://github.com/dlang/dmd/pull/3482#issuecomment-64838230

Since this code won't be compiled in this is just to ensure no diagnostics are emitted.